packages/blade/src/components/Chip/__tests__/Chip.web.test.tsx

```diff
@@ -0,0 +1,138 @@
+import userEvents from '@testing-library/user-event';
+import React from 'react';
+import { Chip } from '../Chip';
+import { ChipGroup } from '../ChipGroup/ChipGroup';
+import renderWithTheme from '~utils/testing/renderWithTheme.web';
+import { Button } from '~components/Button';
+import { InfoIcon } from '~components/Icons';
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+describe('<Chip />', () => {
+  it('should render chip', () => {
+    const { container, getByRole } = renderWithTheme(
+      <ChipGroup>
+        <Chip value="apple">Apple</Chip>
+        <Chip value="mango">Mango</Chip>
+      </ChipGroup>,
+    );
+    expect(container).toMatchSnapshot();
+    expect(getByRole('radio', { name: 'Apple' })).toBeInTheDocument();
+    expect(getByRole('radio', { name: 'Mango' })).toBeInTheDocument();
+  });
+
+  it('should render chip with icon', () => {
+    const { container, getByRole } = renderWithTheme(
+      <ChipGroup>
+        <Chip value="apple" icon={InfoIcon}>
+          Apple
+        </Chip>
+        <Chip value="mango" icon={InfoIcon}>
+          Mango
+        </Chip>
+      </ChipGroup>,
+    );
+    expect(container).toMatchSnapshot();
+    expect(getByRole('radio', { name: 'Apple' })).toBeInTheDocument();
+    expect(getByRole('radio', { name: 'Mango' })).toBeInTheDocument();
+  });
+
+  it('should throw error if used outside ChipGroup', () => {
+    try {
+      renderWithTheme(<Chip value="apple">Apple</Chip>);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        expect(error.message).toMatchInlineSnapshot(
+          `"[Blade: Chip]: <Chip /> component should only be used within the context of a <ChipGroup /> component"`,
+        );
+      }
+    }
+  });
+
+  it('should set disabled state with isDisabled', () => {
+    const { container, getByRole } = renderWithTheme(
+      <ChipGroup>
+        <Chip value="apple" isDisabled>
+          Apple
+        </Chip>
+        <Chip value="mango">Mango</Chip>
+      </ChipGroup>,
+    );
+    expect(container).toMatchSnapshot();
+    expect(getByRole('radio', { name: 'Apple' })).toBeDisabled();
+    expect(getByRole('radio', { name: 'Mango' })).not.toBeDisabled();
+  });
+
+  it('user should be able set checked state', async () => {
```

Comment:
let's add test for selectionType single and multiple.

Reply:
Added, it's covered in ChipGroup tests too.